device/Microsoft.Azure.Devices.Client/DeviceAuthenticationWithSakRefresh.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Azure.Devices.Client.Extensions;
+using System;
+using System.Net;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Devices.Client
+{
+    internal class DeviceAuthenticationWithSakRefresh : DeviceAuthenticationWithTokenRefresh
```

Comment:
you mean to name it SasRefresh, I supposed...

Reply:
Short-name for SharedAccessKey. I noticed that we call SAS Token simply "Token".
I'll add a comment to clarify.